File: src/FoodSharing.Site/ClientApp/src/domain/announcements/announcementCategory.ts
export class AnnouncementCategory {
    constructor(
        public id: string,
        public name: string,
        public iconUrl: string
    ) { }
}

export function mapToAnnouncementCategory(data: any) {
    return new AnnouncementCategory(data.id, data.name, data.iconUrl);
}

File: src/FoodSharing.Site/Models/Announcements/AnnouncementCategory.cs
namespace FoodSharing.Site.Models.Announcements;

public class AnnouncementCategory
{
    public Guid Id { get; }
    public String Name { get; }
    public String IconUrl { get; }

    public AnnouncementCategory(Guid id, String name, String iconUrl)
    {
        Id = id;
        Name = name;
        IconUrl = iconUrl;
    }
}